src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,420 @@
+        for ao_ch, do_ch, crv, pwr in zip(self._ao_channels, self._do_channels, self._pwr_curve, value):
+            pwr = min(pwr, crv[-1][1])
+            volt = self._power_to_volt(pwr, crv)
+            data = int((volt / MAX_VOLTAGE) * MAX_VOLTAGE_VALUE)  # data input expects an uint16
+            # update the analogue output value
+            logging.debug(f"Setting ao_channel {ao_ch} to {volt} V = {pwr} W")
+            self.device.AOut(ao_ch, data)
+
+            port, bit = MCCDevice.channel_to_port(do_ch)
+            old_bit_value = self.device.DBitIn(port, bit)
+            new_bit_value = int(pwr > 0.0)
+            # update the digital output value by using a direct digital port bit
+            if old_bit_value != new_bit_value:
+                logging.debug(f"Setting do_channel {do_ch} from {old_bit_value} to {new_bit_value}")
+                self.device.DBitOut(port, bit, new_bit_value)
+
+
+class MCCDeviceSimulator(object):
+    """
+    A really basic and simple interface to simulate a USB-1208LS device with
+    support for DIO pin read/write commands and a single AO write command.
+    """
+
+    def __init__(self):
+        # initialize values
+        self.productID = 0x0007a  # USB-1208LS
+        # to keep track of the individual bits and values
+        self.port_a_bit_config = {"0": 0, "1": 0, "2": 0, "3": 0, "4": 0, "5": 0, "6": 0, "7": 0}
+        self.port_b_bit_config = {"0": 1, "1": 1, "2": 1, "3": 1, "4": 1, "5": 1, "6": 1, "7": 1}
+        self.AO_channels = {"0": 0, "1": 0}  # value (uint16) in counts to output [10-bits 0-5V]
+
+        # set default configuration
+        self.DConfig(usb_1208LS.DIO_PORTA, 0xff)  # Port A output
+        self.DConfig(usb_1208LS.DIO_PORTB, 0x00)  # Port B input
+        # self.DOut(usb_1208LS.DIO_PORTA, 0x0)
+        # self.DOut(usb_1208LS.DIO_PORTB, 0x0)
+        self.AOut(0, 0x0)
+        self.AOut(1, 0x0)
+
+    def getManufacturer(self):
+        return "MCC"
+
+    def getProduct(self):
+        return "USB-1208LS"
+
+    def getSerialNumber(self):
+        return "021B0CB8"
+
+    def DConfig(self, port_number, bit_mask):
+        """
+        This command sets the direction of the digital bits for a port.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit_mask (int:bit value): 0 = output | 1 = input
+        """
+        if self.productID == 0x0075 and port_number == usb_1208LS.DIO_AUXPORT:
+            bit_mask = ((bit_mask ^ 0xff) & 0xff)
+
+        self.DOut(port_number, bit_mask)
+
+    def DIn(self, port_number):
+        """
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :return: the value seen at the port pins
+        """
+        ret_val = "0b"
+        DIO_port = self._return_port_config(port_number)
+
+        # use a sorted dict to be sure of the key order
+        for bit in list(sorted(DIO_port.keys()))[::-1]:
+            ret_val += str(DIO_port[bit])
```

Comment:
Assuming it's a list, you can just write:
`return sum(v << i for i, v in enumerate(DIO_port))`